crates/biome_graphql_analyze/src/lint/nursery/no_duplicated_graphql_fields.rs

```diff
@@ -0,0 +1,198 @@
+use std::collections::HashSet;
+
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_graphql_syntax::AnyGraphqlOperationDefinition;
+use biome_graphql_syntax::AnyGraphqlSelection;
+use biome_graphql_syntax::GraphqlArguments;
+use biome_graphql_syntax::GraphqlSelectionSet;
+use biome_graphql_syntax::GraphqlVariableDefinitions;
+use biome_rowan::AstNode;
+use biome_rowan::TextRange;
+
+declare_rule! {
+    /// No duplicated fields in GraphQL operations.
+    ///
+    /// Checks for duplicate fields in selection set, variables in operation definition, or in arguments set of a field.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```graphql,expect_diagnostic
+    /// query {
+    ///   user {
+    ///     id
+    ///     id
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```graphql
+    /// query {
+    ///   user {
+    ///     id
+    ///   }
+    /// }
+    /// ```
+    ///
+    pub NoDuplicatedGraphqlFields {
+        version: "next",
+        name: "noDuplicatedGraphqlFields",
+        language: "graphql",
+        recommended: true,
```

Comment:
Since this rule comes from an existing one, you should add the source: https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#biome-lint-rules-inspired-by-other-lint-rules

Not sure if we should do the same for the other rule

Reply:
Ah yes, I forgot about that. The other rule also has an equivalent version in graphql-eslint, I guess it makes sense to also add a source for it